src/components/DoctorDetails.js

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useNavigate, useParams } from 'react-router-dom';
+import {
+  doctorDetailsSelector,
+  fetchDoctorDetails,
+} from '../redux/doctors/detailsSlice';
+import DetailsTable from './DetailsTable';
+import '../styles/doctor-details.css';
+import { deleteDoctor } from '../service/doctorsService';
+
+const DoctorDetails = () => {
```

Comment:
- [ ] When I try to delete a doctor that has a reservation, this happens:

1- I click on the "Delete" button:

![image](https://github.com/0sugo/appointment_booker_frontend/assets/57421823/2ed51c36-e31e-43a1-ac5b-4ca6eaaefb34)

2- It fails with a 500:

![image](https://github.com/0sugo/appointment_booker_frontend/assets/57421823/e7da46ce-b28d-4f10-86d7-ea7855cc5ea0)

Console:

![image](https://github.com/0sugo/appointment_booker_frontend/assets/57421823/515931b3-8a95-4912-bb48-24e10c8be647)

This is because the doctor in question still has an appointment and there is a constraint so that if a doctor still exists on the reservations table, you won't be able to delete it.

Therefore, I leave this to you and you have 2 options:

1- Make this possible (maybe you fired a doctor or he/she quitted).  This can be done by adding `dependant: destroy` in the doctor's model:

```ruby
has_many :reservations, dependent: :destroy
```
Look:
![image](https://github.com/0sugo/appointment_booker_frontend/assets/57421823/46c5be81-415a-4858-aefa-225f69a11dc4)

2- Leave it like this, if this is a feature.  Just kindly give some sort of message to your users in the frontend explaining why the deletion action failed.

That way, the UX will be improved. 💪